### PR TITLE
Generate decks at local execution

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
this [PR](https://github.com/flyteorg/flytekit/commit/06d95b805d36998b58cce93f3837f0bd0ec0b945) breaks it, because it moves `_outpu_deck` to `entrypoint.py` . we use `entrypoint.py` only when we run the workflow remotely.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
https://unionai.slack.com/archives/C02CMUNT4PQ/p1687268923664029

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by samhita-alla** [comment]
@pingsutw, some checks are failing. Can you take a look at them?

**Review by wild-endeavor — APPROVED** [review]
what happens if the same task is run twice in the same workflow?  does it get overwritten?

**Comment by pingsutw** [comment]
No, The deck files will be written to different locations.
